src/main/java/com/cloudogu/scm/favorite/RepositoryLinkEnricher.java

```diff
@@ -0,0 +1,69 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.cloudogu.scm.favorite;
+
+import sonia.scm.api.v2.resources.Enrich;
+import sonia.scm.api.v2.resources.HalAppender;
+import sonia.scm.api.v2.resources.HalEnricher;
+import sonia.scm.api.v2.resources.HalEnricherContext;
+import sonia.scm.api.v2.resources.LinkBuilder;
+import sonia.scm.api.v2.resources.ScmPathInfoStore;
+import sonia.scm.plugin.Extension;
+import sonia.scm.repository.Repository;
+import sonia.scm.repository.RepositoryPermissions;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+@Extension
+@Enrich(Repository.class)
+public class RepositoryLinkEnricher implements HalEnricher {
+
+  private final Provider<ScmPathInfoStore> scmPathInfoStore;
+  private final FavoriteRepositoryProvider storeProvider;
+
+  @Inject
+  public RepositoryLinkEnricher(Provider<ScmPathInfoStore> scmPathInfoStore, FavoriteRepositoryProvider storeProvider) {
+    this.scmPathInfoStore = scmPathInfoStore;
+    this.storeProvider = storeProvider;
+  }
+
+  @Override
+  public void enrich(HalEnricherContext context, HalAppender appender) {
+    Repository repository = context.oneRequireByType(Repository.class);
+    if (RepositoryPermissions.read().isPermitted(repository)) {
+      LinkBuilder linkBuilder = new LinkBuilder(scmPathInfoStore.get().get(), FavoriteRepositoryResource.class);
+      FavoriteRepository favorites = storeProvider.get().get();
+      if (favorites != null && favorites.isFavorite(repository)) {
+        appender.appendLink("unfavorize", linkBuilder.method("unfavorizeRepository").parameters(repository.getNamespace(), repository.getName()).href());
+      } else {
+        appender.appendLink("favorize", linkBuilder.method("favorizeRepository").parameters(repository.getNamespace(), repository.getName()).href());
+      }
+      appender.linkArrayBuilder("favorites")
+        .append("unfavorize", linkBuilder.method("unfavorizeRepository").parameters(repository.getNamespace(), repository.getName()).href())
+        .append("favorize", linkBuilder.method("favorizeRepository").parameters(repository.getNamespace(), repository.getName()).href())
+        .build();
+    }
```

Comment:
Because we append the single link, so we know if the repository is currently favorized or not. We need both links if we want to toggle the star without pagereload. Otherwise the toggling would not work, since the second link is missing.